pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -106,14 +119,31 @@ effectiveStdenv.mkDerivation (finalAttrs: {
   env.NIX_LDFLAGS = lib.concatStringsSep " " (finalAttrs.darwinLdFlags ++ finalAttrs.metalLdFlags);
 
   env.NIX_CFLAGS_COMPILE =
-    lib.optionalString (march != "") "-march=${march}" + lib.optionalString (mtune != "") "-mtune=${mtune}";
+    let
+      inherit (stdenv.hostPlatform)
+        isx86_64
+        isAarch64
+        isLinux
+        isDarwin
+        ;
+    in
+    lib.concatStringsSep " " [
+      (lib.optionalString isx86_64 "-march=x86-64")
+      (lib.optionalString (isAarch64 && isLinux) "-march=armv8.7-a")
+      (lib.optionalString (mtune != "") "-mtune=${mtune}")
+      # https://stackoverflow.com/a/69926065
+      (lib.optionalString isDarwin "-mcpu=apple-m1")
```

Comment:
Similarly, if it's important to provide a build targetting specifically this instruction set maybe we could expose in a custom attribute, but ignoring the hostPlatform configuration is a default that would require some justification. Does the build fail without this flag? Is there a verifiable difference in the performance with and without this flag?

Thanks